app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/mapper/SwapWidgetMapper.kt

```diff
@@ -0,0 +1,138 @@
+package org.p2p.wallet.swap.ui.jupiter.main.mapper
+
+import org.p2p.core.common.TextContainer
+import org.p2p.core.utils.formatFiat
+import org.p2p.core.utils.formatToken
+import org.p2p.uikit.utils.skeleton.textCellSkeleton
+import org.p2p.uikit.utils.text.TextViewCellModel
+import org.p2p.uikit.utils.toPx
+import org.p2p.wallet.R
+import org.p2p.wallet.swap.jupiter.domain.model.SwapTokenModel
+import org.p2p.wallet.swap.ui.jupiter.main.widget.SwapWidgetModel
+import java.math.BigDecimal
+
+class SwapWidgetMapper {
+
+    fun mapWidgetLoading(isTokenA: Boolean): SwapWidgetModel {
+        return if (isTokenA) {
+            SwapWidgetModel.Loading(
+                isStatic = false,
+                widgetTitle = TextViewCellModel.Raw(text = TextContainer(R.string.swap_main_you_pay)),
+            )
+        } else {
+            SwapWidgetModel.Loading(
+                isStatic = true,
+                widgetTitle = TextViewCellModel.Raw(text = TextContainer(R.string.swap_main_you_receive)),
+            )
+        }
+    }
+
+    fun mapTokenA(
+        token: SwapTokenModel,
+        tokenAmount: BigDecimal? = null,
+    ): SwapWidgetModel {
+        return SwapWidgetModel.Content(
+            isStatic = false,
+            widgetTitle = swapWidgetFromTitle(),
+            availableAmount = availableAmount(token),
+            balance = balance(token),
+            currencyName = tokenName(token),
+            amount = tokenAmount(token, tokenAmount),
+            fiatAmount = fiatAmount(token, tokenAmount),
+            amountMaxDecimals = token.decimals,
+        )
+    }
+
+    fun mapTokenB(token: SwapTokenModel, tokenAmount: BigDecimal?): SwapWidgetModel {
+        var fiatAmount = fiatAmount(token, tokenAmount)
+        if (true) {
+            // todo price impact
+            fiatAmount = fiatAmount?.copy(textColor = R.color.text_night)
+        }
+        return SwapWidgetModel.Content(
+            isStatic = true,
+            widgetTitle = swapWidgetToTitle(),
+            availableAmount = null,
+            balance = balance(token),
+            currencyName = tokenName(token),
+            amount = tokenAmount(token, tokenAmount),
+            fiatAmount = fiatAmount,
+            amountMaxDecimals = token.decimals,
+        )
+    }
+
+    fun mapTokenBLoading(
+        token: SwapTokenModel,
+    ): SwapWidgetModel.Content =
+        SwapWidgetModel.Content(
+            isStatic = true,
+            widgetTitle = swapWidgetToTitle(),
+            currencyName = tokenName(token),
+            amount = textCellSkeleton(
+                height = 20.toPx(),
+                width = 84.toPx(),
+                radius = 6f.toPx(),
```

Comment:
Are you sure we should have static height for this?

Reply:
yes it is less then text size height